swap_meet/vendor.py

```diff
@@ -1,2 +1,62 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        result = item
+        self.inventory.append(result)
+        return result
+    
+    def remove(self, item):
+        result = item
+        if item in self.inventory:
+            self.inventory.remove(result)
+        else:
+            return False
+        return result
+
+    def get_by_category(self, category):
+        items_by_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        return items_by_category
+
+    def swap_items(self, friend, my_item, their_item):
+        if not self.remove(my_item):
+            return False
+        if not friend.remove(their_item):
+            self.add(my_item)
+            return False
```

Comment:
Consider including this in one compound conditional to enhance readability.

```suggestion
        if not self.remove(my_item) or not friend.remove(their_item):
            return False
```